File: Lainnya/movie-catalogue-jetpack-2-main/movie-catalogue-jetpack-2-main/movie-catalogue-jetpack-2-main/app/src/main/java/com/mramirid/moviecatalogue/di/Injection.java
package com.mramirid.moviecatalogue.di;

import com.mramirid.moviecatalogue.data.source.MovieCatalogueRepository;
import com.mramirid.moviecatalogue.data.source.remote.RemoteRepository;
import com.mramirid.moviecatalogue.utils.JsonHelper;

public class Injection {

	public static MovieCatalogueRepository provideRepository() {
		RemoteRepository remoteRepository = RemoteRepository.getInstance(new JsonHelper());
		return MovieCatalogueRepository.getInstance(remoteRepository);
	}
}

File: Lainnya/cielo/JetpackSubmission3v2/app/src/main/java/com/gracielo/jetpacksubmission3v2/DataInjection/Injection.kt
package com.gracielo.jetpacksubmission3v2.DataInjection

import android.content.Context
import com.gracielo.jetpacksubmission3v2.Data.Local.LocalDataSource
import com.gracielo.jetpacksubmission3v2.Data.Local.Room.FilmDatabase
import com.gracielo.jetpacksubmission3v2.Data.Source.MovieCatalogueRepository
import com.gracielo.jetpacksubmission3v2.Data.Source.Remote.RemoteRepository
import com.gracielo.jetpacksubmission3v2.Utils.AppExecutors
import com.gracielo.jetpacksubmission3v2.Utils.JsonHelper

object Injection {
    fun provideRepository(context: Context): MovieCatalogueRepository {

        val database = FilmDatabase.getInstance(context)

        val remoteDataSource = RemoteRepository.getInstance(JsonHelper(context))
        val localDataSource = LocalDataSource.getInstance(database.filmDao())
        val appExecutors = AppExecutors()
        return MovieCatalogueRepository.getInstance(remoteDataSource!!,localDataSource,appExecutors)
    }
}